app/controllers/lottery_controller.rb

```diff
@@ -9,31 +9,52 @@ def action_allowed?
   def run_intelligent_assignment
     json_info = []
     #get topics for assignment
-    topics = SignUpTopic.where(assignment_id: params[:id])
-    topics.each do |t|
-      t = t.id
-    end
+    topics = SignUpTopic.where(assignment_id: params[:id]).map{ |t| t.id}
     # students is a list of student ids in an assignment
-    students = Participant.where(parent_id: params[:id])
+    students = Participant.where(parent_id: params[:id]).map{ |p| p.user_id}
     students.each do |student|
       #grab student id and list of bids
-      s = student.id
-      bids = Bid.where(team_id: s).sort_by{ |b| [topics.index(b.topic_id)]}
-      json_info << {"pid"=>s,"ranks"=>bids.map{|b| b.priority}}
+      bids = Bid.where(team_id: student).map{|b| {priority:b.priority,topic_id:b.topic_id}}.sort_by{ |b| [topics.index(b[:topic_id])]}
+      topics.each do |topic|
+        if !bids.any? {|b| b[:topic_id] == topic}
+          bids.insert(topics.index(topic),{priority:0,topic_id:topic})
+        end
+      end
+      json_info << {"pid"=>student,"ranks"=>bids.map{|b| b[:priority]}}
     end
-    json_data = {"users"=>json_info,"max_team_size"=>Assignment.find_by_id(params[:id]).max_team_size}
+    json_data = {'Content-Type' => 'application/json',"users"=>json_info,"max_team_size"=>Assignment.find_by_id(params[:id]).max_team_size}
     #send json_data with a get request and get teams
-    response = Net::HTTP.post_form(URI("http://peerlogic.csc.ncsu.edu/intelligent_assignment/merge_teams/"), json_data)
-    puts "a;ldfja;lfjalk;sfjal;kjfl;kasjfl;asjdfl;ajsdf;lajsdf;lajsdf;lkajs;ldfja;sldfja;lsdjfa;lsdjf;laksdjf"
-    puts response.body
+    uri = URI.parse("http://peerlogic.csc.ncsu.edu/intelligent_assignment/merge_teams")
+    http = Net::HTTP.new(uri.host, uri.port)
+    request = Net::HTTP::Post.new(uri.request_uri, 
+          'Content-Type' => 'application/json')
+    request.body = json_data.to_json()
+    response = http.request(request)
     teams = JSON.parse(response.body)["teams"]
     teams.each do |team|
-      captain = TeamsUser.where(user_id: team[0]).team_id
+      t = nil
+      #find if existing team exists
+      team.each do |student|
+        TeamsUser.where(user_id: student).each do |user|
+          nTeam = Team.where(id: user.team_id, parent_id: params[:id]).first
+          if t.nil? and !nTeam.nil?
+            t = nTeam
+          end
+        end
+      end
+      if t.nil?
+        #create the team
```

Comment:
#--zhewei-----06/22/2015---------
    # If you want to create a new team with topic and team members on view, you have to
    # 1. create new Team
    # 2. create new TeamsUser
    # 3. create new SignedUpTeam
    # 4. create new TeamNode (node_object_id of TeamNode is team_id)
    # 5. create new TeamUserNode (node_object_id of TeamUserNode is teams_user_id)
And you can search for `create_new_team` method.